indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/PemEncodable.kt

```diff
@@ -0,0 +1,110 @@
+package at.asitplus.signum.indispensable.asn1
+
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+
+/**
+ * Specialization of [Asn1Encodable], able to produce PEM-encoded strings
+ * as per [RFC 1421](https://datatracker.ietf.org/doc/html/rfc1421#section-4.4).
+ * Use in tandem with [PemDecodable].
+ */
+interface PemEncodable<A : Asn1Element> : Asn1Encodable<A> {
+
+    /**
+     * Encapsulation boundary string.
+     * Will be automatically fenced.
+     * Make sure it matches the corresponding [PemDecodable.ebString]!
```

Comment:
I wonder if we could create a test case for this using reflection... iterate over all implementers of `PemEncodable`, instantiate, get `ebString`, get companion object, assert `PemDecodable`, get `ebString`... similar to what #175 does to assert the fake-enum entries for completeness.

Reply:
that would be nice. i'd to that in conjuction with making everything else PEM encodable that is PEM encodable